### PR TITLE
ENCORE - replace align.rms_fit_trj with align.AlignTraj(...).run()

### LINKED ISSUE #1098: ENCORE code and docs still use deprecated align.rms_fit_trj()
The examples for [Functions for ensemble comparisons](http://www.mdanalysis.org/mdanalysis/documentation_pages/analysis/encore/similarity.html#functions-for-ensemble-comparisons)
 still use the deprecated `align.rms_fit_trj()` function.  The code itself (`MDAnalysis/analysis/encore/similarity.py`) and the tests (`testsuite/MDAnalysisTests/analysis/test_encore.py`) also use it.

Instead, use `align.AlignTraj(...).run()` (see #1059).

- [ ] ENCORE code updated
- [ ] ENCORE docs updated

### PR DESCRIPTION
**Pull request by mtiberti**
Fixes #1098 

Changes made in this Pull Request:
 - replaced rms_fit_trj with AlignTraj(...).run() in similarity.py
 - replaced rms_fit_trj with AlignTraj(...).run() in tests (and slightly streamlined two of them)
 - updated the docs to reflect the changes

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/encore/similarity.py [hunk 1]
```diff
@@ -861,10 +861,10 @@ def hes(ensembles,
     # on the universe.
     if align:
         for ensemble in ensembles:
-            mda.analysis.align.rms_fit_trj(ensemble, ensembles[0],
+            mda.analysis.align.AlignTraj(ensemble, ensembles[0],
                                            select=selection,
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/encore/similarity.py]: Minor remark. The indentation is now off by 2 spaces here. If you could change it here and for the other parameters as well that would be nice.

Comment (mtiberti) [inline package/MDAnalysis/analysis/encore/similarity.py]: sure - just did it in similarity.py and also in the test file

File: testsuite/MDAnalysisTests/analysis/test_encore.py [hunk 2]
```diff
@@ -223,31 +207,21 @@ def test_ensemble_superimposition():
 
     @staticmethod
     def test_ensemble_superimposition_to_reference_non_weighted():
-        ensemble0 = mda.Universe(PSF, DCD)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_encore.py]: Is there a reason why you removed the ensemble0 test

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_encore.py]: coveralls reports that coverage has decreased. That is strange for such a minimal change.

Comment (mtiberti) [inline testsuite/MDAnalysisTests/analysis/test_encore.py]: ensemble0 was actually not being used later on in the assert_equal - it was probably some leftover for debugging purposes, so it didn't make much sense to keep it

**Comment by kain88-de** [comment]
the coverage drop is in deprecated code. So it's all fine.

**Pull request merged**